### ISSUE TITLE
`conda-build` 3.19.2 is installed but unable to run it. Instead 3.18.11 is what gets called.

### ISSUE THREAD

**Issue by 00krishna** [conda/conda-build#4006]
### Actual Behavior

I created a `conda` environment and installed version 3.19.2 of `conda-build`. But, when I check the `conda build --version` I get 3.18.11 instead of the newer version. I need to use a version of `conda-build`>3.18.12 in order to get past issue https://github.com/conda/conda-build/issues/3956 and these CUDE incompatibility issues.

### Expected Behavior

I should be able to install `conda-build` 3.19.2, and then that version will be used to implement the build process.

### Steps to Reproduce

1. create a new anaconda environment using `conda` version 4.8.3. I am running Ubuntu linux 18.04 LTS.
```
>>>conda create -c conda-forge -n pkg-builder conda-build
>>>conda activate pkg-builder
```
2. When I check the conda-build version it indicates 3.19.2
```
 >>>conda list conda-build

 Name                    Version                   Build  Channel \
 conda-build               3.19.2           py38h32f6830_2    conda-forge
```
3. When I try and use conda-build it reads an older version
```
>>>conda build --version

3.18.11
```
This last part is the problem. When I try and run `conda build .` to build a package, the computer will run version 3.18.11 of `conda build` instead of the newer version. And I need the newer version otherwise I end up with extra dependencies in the package that breaks the python `noarch` architecture of the package.

##### `Output of conda info`

```
conda info

     active environment : test-package
    active env location : /media/krishnab/lakshmi/anaconda3/envs/test-package
            shell level : 2
       user config file : /home/krishnab/.condarc
 populated config files : /home/krishnab/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.8.3.final.0
       virtual packages : __cuda=10.0
                          __glibc=2.27
       base environment : /media/krishnab/lakshmi/anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /media/krishnab/lakshmi/anaconda3/pkgs
                          /home/krishnab/.conda/pkgs
       envs directories : /media/krishnab/lakshmi/anaconda3/envs
                          /home/krishnab/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.24.0 CPython/3.8.3 Linux/4.15.0-112-generic ubuntu/18.04.4 glibc/2.27
                UID:GID : 1000:1000
             netrc file : /home/krishnab/.netrc
           offline mode : False

```

The output of `conda list conda-build`:

```
# Name                    Version                   Build  Channel
conda-build               3.19.3+103.g2b2898fb          pypi_0    pypi
```

Check the version of the `conda build ` package from the executable. 
```
>>>conda build --version
conda-build 3.18.11
```

**Comment by 00krishna** [comment]
This also seems to be a related issue.
https://github.com/KevinMusgrave/pytorch-metric-learning/issues/55

**Comment by 00krishna** [comment]
Okay, I was able to solve this. The problem seems to have been with my `condarc` file. When I changed from `strict` priority to `flexible` priority, the installation of `conda-build` 3.19.2 worked just fine. So apparently there are some dependencies for `conda-build` in the `defaults` channel that are not in the `conda-forge` channel. I know that we are encouraged not to mix packages from different channels, so this might warrant some attention.
